src/EnergyPlus/SteamCoils.cc

```diff
@@ -1140,55 +1139,183 @@ namespace SteamCoils {
-                    state.dataSteamCoils->SteamCoil(CoilNum).OperatingCapacity = QSteamCoilMaxHT;
+                SetComponentFlowRate(state,
+                                     SteamMassFlowRate,
+                                     state.dataSteamCoils->SteamCoil(CoilNum).SteamInletNodeNum,
+                                     state.dataSteamCoils->SteamCoil(CoilNum).SteamOutletNodeNum,
+                                     state.dataSteamCoils->SteamCoil(CoilNum).plantLoc);
 
-                    // Determine the Max coil capacity and check for the same.
-                    if (QCoilReq > QSteamCoilMaxHT) {
-                        QCoilCap = QSteamCoilMaxHT;
-                    } else {
-                        QCoilCap = QCoilReq;
-                    }
+                // recalculate if mass flow rate changed in previous call.
+                QCoilCap = SteamMassFlowRate * (LatentHeatSteam + SubcoolDeltaTemp * CpWater);
 
-                    // Steam Mass Flow Rate Required
-                    SteamMassFlowRate = QCoilCap / (LatentHeatSteam + SubcoolDeltaTemp * CpWater);
+                // In practice Sensible & Superheated heat transfer is negligible compared to latent part.
+                // This is required for outlet water temperature, otherwise it will be saturation temperature.
+                // Steam Trap drains off all the Water formed.
+                // Here Degree of Subcooling is used to calculate hot water return temperature.
+
+                // Calculating Water outlet temperature
+                TempWaterOut = TempSteamIn - SubcoolDeltaTemp;
+
+                // Total Heat Transfer to air
+                HeatingCoilLoad = QCoilCap;
+
+                // Temperature of air at outlet
+                TempAirOut = TempAirIn + QCoilCap / (AirMassFlow * PsyCpAirFnW(Win));
+
+                state.dataSteamCoils->SteamCoil(CoilNum).OutletSteamMassFlowRate = SteamMassFlowRate;
+                state.dataSteamCoils->SteamCoil(CoilNum).InletSteamMassFlowRate = SteamMassFlowRate;
+
+                //************************* Loop Losses *****************************
+                // Loop pressure return considerations included in steam coil since the pipes are
+                // perfect and do not account for losses.
+                // Return water is condensate at atmoshperic pressure
+                // Process is considered constant enthalpy expansion
+                // No quality function in EnergyPlus hence no option left apart from
+                // considering saturated state.
+                //              StdBaroPress=101325
+
+                TempWaterAtmPress = GetSatTemperatureRefrig(
+                    state, fluidNameSteam, state.dataEnvrn->StdBaroPress, state.dataSteamCoils->SteamCoil(CoilNum).FluidIndex, RoutineName);
+
+                // Point 4 at atm - loop delta subcool during return journery back to pump
```

Comment:
Double spaced with each line of code commented. Like a CS 101 homework assignment.